lib/domain/contact_manager/contacts_manager.dart

```diff
@@ -31,55 +26,42 @@ class ContactsManager {
   final PermissionHandlerService _permissionHandlerService =
       PermissionHandlerService();
 
-  Set<Contact> _tomContacts = {};
-
-  Set<Contact> _phonebookContacts = {};
+  final _phonebookContactInteractor = getIt.get<PhonebookContactInteractor>();
 
-  Set<Contact> get tomContacts => _tomContacts;
+  ValueNotifier<Either<Failure, Success>> contactsNotifier =
+      ValueNotifier(const Right(ContactsInitial()));
 
-  Set<Contact> get phonebookContacts => _phonebookContacts;
+  ValueNotifier<Either<Failure, Success>> phonebookContactsNotifier =
+      ValueNotifier(const Right(GetPhonebookContactsInitial()));
 
-  bool get _firstSynchronizing =>
-      _syncContactsState == SyncContactsState.synchronizing ||
-      _syncContactsState == SyncContactsState.initial;
+  ValueNotifier<WarningContactsBannerState> warningBannerNotifier =
+      ValueNotifier(WarningContactsBannerState.hide);
 
-  bool _isExternalContact(String keyword) =>
-      keyword.isValidMatrixId && keyword.startsWith("@");
-
-  StreamController<Either<Failure, Success>> contactsStream =
-      StreamController<Either<Failure, Success>>.broadcast();
+  ContactsManager() {
+    FlutterContacts.addListener(_fetchPhonebookContacts);
```

Comment:
Why do you use `FlutterContacts` in here? 
IMO: It only handles contacts from `datt` layer move it to `PhonebookApi`

Reply:
any better idea ?